processing/src/main/java/org/apache/druid/query/metadata/SegmentAnalyzer.java

```diff
@@ -192,6 +193,7 @@ private ColumnAnalysis analyzeNumericColumn(
     return new ColumnAnalysis(
         capabilities.getType().name(),
         capabilities.hasMultipleValues().isTrue(),
+        capabilities.isNullable().isTrue(),
```

Comment:
I think maybe the only current unknowns will come from complex columns, which thinking about it should probably just report that they have nulls as true instead of unknown...

Right now nothing bad would happen if the segment metadata falsely reports not having nulls, but in the follow-up where this is wired into `DruidSchema`, falsely reporting nulls would cause the SQL planner to incorrectly plan to not handle nulls, so I think this should probably be modified to check `isMaybeTrue`.